hs_core/management/commands/migrate_modflow_instance_resources.py

```diff
@@ -0,0 +1,351 @@
+                self.stdout.flush()
+                # skip this mi resource
+                continue
+
+            # change the resource_type
+            mi_metadata_obj = mi_res.metadata
+            mi_res.resource_type = to_resource_type
+            mi_res.content_model = to_resource_type.lower()
+            mi_res.save()
+            # get the converted resource object - CompositeResource
+            comp_res = mi_res.get_content_model()
+
+            # set CoreMetaData object for the composite resource
+            core_meta_obj = CoreMetaData.objects.create()
+            comp_res.content_object = core_meta_obj
+
+            # migrate mi resource core metadata elements to composite resource
+            migrate_core_meta_elements(mi_metadata_obj, comp_res)
+            comp_res.save()
+
+            # update url attribute of the metadata 'type' element
+            type_element = comp_res.metadata.type
+            type_element.url = '{0}/terms/{1}'.format(current_site_url(), to_resource_type)
+            type_element.save()
+
+            # create a mi aggregation
+            mi_aggr = ModelInstanceLogicalFile.create(resource=comp_res)
+            mi_aggr.save()
+            try:
+                move_files_and_folders_to_model_aggregation(command=self, model_aggr=mi_aggr, comp_res=comp_res,
+                                                            logger=logger, aggr_name='modflow-model-instance')
+            except Exception as ex:
+                err_resource_counter += 1
+                err_msg = '{}Failed to move files/folders into model instance aggregation for resource (ID: {})'
+                err_msg = err_msg.format(self._MIGRATION_ISSUE, comp_res.short_id)
+                err_msg = err_msg + '\n' + str(ex)
+                logger.error(err_msg)
+                self.stdout.write(self.style.ERROR(err_msg))
+                self.stdout.flush()
+                mi_metadata_obj.delete()
+                continue
+            # copy the resource level keywords to aggregation level
+            if comp_res.metadata.subjects:
+                keywords = [sub.value for sub in comp_res.metadata.subjects.all()]
+                mi_aggr.metadata.keywords = keywords
+                mi_aggr.metadata.save()
+            # copy the model specific metadata to the mi aggregation
+            if mi_metadata_obj.model_output:
+                mi_aggr.metadata.has_model_output = mi_metadata_obj.model_output.includes_output
+
+            if self._EXECUTED_BY_EXTRA_META_KEY in comp_res.extra_data:
+                if not set_executed_by(self, mi_aggr, comp_res, logger):
+                    err_resource_counter += 1
+
+            mi_aggr.save()
+            # load the default MODFLOW instance meta schema - Note: this schema is used only for migration and
+            # is not a template schema that the user can select using the UI for metadata editing
+            meta_json_schema = get_modflow_meta_schema()
```

Comment:
Is there a model program for the modflow schema?  I only see model instances created.

Reply:
The existing modflow specific metadata in MODFLOW model instance resource needs to be stored as json data in the model instance aggregation being created as part of the migration. The aggregation also needs to have a schema so that the aggregation json metadata can be edited.